packages/design-system-react-native/src/primitives/TextOrChildren/TextOrChildren.test.tsx

```diff
@@ -0,0 +1,27 @@
+import { render } from '@testing-library/react-native';
+import React from 'react';
+
+import Text from '../../components/Text';
+import TextOrChildren from './TextOrChildren';
+
+describe('TextOrChildren', () => {
+  it('renders Text component when children is a string', () => {
+    const { getByText } = render(<TextOrChildren>Sample Text</TextOrChildren>);
+    expect(getByText('Sample Text')).toBeDefined();
+  });
+
+  it('renders child components when children is not a string', () => {
+    const { getByText } = render(
+      <TextOrChildren>
+        <Text>Nested Text</Text>
+      </TextOrChildren>,
+    );
+
+    expect(getByText('Nested Text')).toBeDefined();
+  });
+
+  it('renders nothing when children is null', () => {
+    const { toJSON } = render(<TextOrChildren>{null}</TextOrChildren>);
+    expect(toJSON()).toBeNull();
+  });
```

Comment:
non-blocking: Not sure if you're using this test for coverage but since `children` is a required prop in `TextOrChildren`, we might want to consider removing this `null` test case. It tests a scenario that TypeScript would prevent at compile time.